platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/internal/artifacts/configurations/DefaultConfigurationExtendsFromIntegrationTest.groovy

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.internal.artifacts.configurations
+
+import org.gradle.api.internal.artifacts.configurations.DefaultConfiguration
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.util.GradleVersion
+
+/**
+ * Integration tests for {@code extendsFrom} method in {@link DefaultConfiguration}.
+ */
+class DefaultConfigurationExtendsFromIntegrationTest extends AbstractIntegrationSpec {
+    def "extending a configuration in another project is deprecated"() {
+        given:
+        settingsFile """
+            include ":project1", ":project2"
+        """
+
+        groovyFile(file('project1/build.gradle'), """
+            configurations {
```

Comment:
In alex's presentation, he said his changes were going to make `groovyFile` not have Gradle-API-specific highlighting. After his changes merge, the highlights in this will go away. How should we handle this? Wait for his changes and use his new `buildFile` overload? Use the non-highlighting syntax? Merge now and accept that we will not get proper highlighting after this merges?

Also, we can avoid `file('')` here and use the `groovyFile(String, String)` overload which is less noisy

Reply:
There's only like a couple dozen usages of `grovyFile`.  I think we merge now and count on Alex to update usages when he changes the behavior.  I'll remove `file()`.